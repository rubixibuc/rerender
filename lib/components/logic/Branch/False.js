import React from "react";
import PropTypes from "prop-types";

const render = ({ children }) => <React.Fragment>{children}</React.Fragment>;

render.propTypes = {};

export default render;
