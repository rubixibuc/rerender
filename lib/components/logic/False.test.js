import React from "react";
import renderer from "react-test-renderer";

import False from "./False";

test("should not render children when children are undefined", () => {
  const component = renderer.create(<False />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("should not render children when children are null", () => {
  const component = renderer.create(<False>{null}</False>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render children when children are defined", () => {
  const component = renderer.create(
    <False>
      <div />
    </False>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
