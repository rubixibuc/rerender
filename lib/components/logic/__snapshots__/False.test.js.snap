// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render children when children are null 1`] = `null`;

exports[`should not render children when children are undefined 1`] = `null`;

exports[`should render children when children are defined 1`] = `<div />`;
